### DISCUSSION TITLE
Compatibility error when installing CLN library with home-brew --- Mac M2 Ventura 13.4

### DISCUSSION THREAD

**Discussion by MicheleD1999** [Homebrew/discussions#5054] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: b3751bca8c4a3c76107d5aa3b75b81db93cf4bb6
Last commit: 7 days ago
Core tap JSON: 14 Jan 10:29 UTC
Core cask tap JSON: 14 Jan 10:29 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 15.1.
```

### Description of issue

Hello to everybody. I'm writing in this forum because I'm kinda new with Mac environment and I did not manage to find a solution alone.

My issue is about the installation of cln library for C++. I first installed brew environment that currently is in /opt/homebrew/bin/brew. Then I launched 

brew install cln

And I got this error after the make launch:

Last 15 lines from /Users/micheledradi/Library/Logs/Homebrew/cln/02.make:
       ^
./base/cl_low.h:210:21: note: declared with C language linkage here
  extern "C" uint32 mulu32_high;                        // -> High-Teil
                    ^
base/low/cl_low_mul.cc:57:8: error: declaration of 'mulu64_high' in global scope conflicts with declaration with C language linkage
uint64 mulu64_high;
       ^
./base/cl_low.h:354:21: note: declared with C language linkage here
  extern "C" uint64 mulu64_high;                        // -> High-Teil
                    ^
2 errors generated.
make[1]: *** [base/low/cl_low_mul.lo] Error 1
libtool: compile:  clang++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I../autoconf -I../include -I../src -I../include -I../src -g -O2 -c base/output/cl_output_dec.cc -o base/output/cl_output_dec.o >/dev/null 2>&1
libtool: compile:  clang++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I../autoconf -I../include -I../src -I../include -I../src -g -O2 -c base/output/cl_output_hex.cc -o base/output/cl_output_hex.o >/dev/null 2>&1
make: *** [all-recursive] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting
Error: A newer Command Line Tools release is available.
Update them from Software Update in System Settings.
If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install
Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 15.1.

As a consequence, I tried to look for Command Line Tools from  https://developer.apple.com/download/all/ but the installation could not be completed because it said that my disk has not macOS 13.5 or newer.

I also tried forcing the linux architecture by adding arch -arm64 before brew install, but I obtained the same error. I really don't know how to proceed.

Thanks to everyone,
Michele

**Comment by SMillerDev** [comment]
You should really update your macOS version to the latest available and then a CLT update should be possible.

**Comment by MicheleD1999** [comment]
I did it but the error still persist, even if now it does not suggest me to update Command Line Tools:


Last 15 lines from /Users/micheledradi/Library/Logs/Homebrew/cln/02.make:
       ^
./base/cl_low.h:210:21: note: declared with C language linkage here
  extern "C" uint32 mulu32_high;                        // -> High-Teil
                    ^
base/low/cl_low_mul.cc:57:8: error: declaration of 'mulu64_high' in global scope conflicts with declaration with C language linkage
uint64 mulu64_high;
       ^
./base/cl_low.h:354:21: note: declared with C language linkage here
  extern "C" uint64 mulu64_high;                        // -> High-Teil
                    ^
2 errors generated.
make[1]: *** [base/low/cl_low_mul.lo] Error 1
libtool: compile:  clang++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I../autoconf -I../include -I../src -I../include -I../src -g -O2 -c base/output/cl_output_dec.cc -o base/output/cl_output_dec.o >/dev/null 2>&1
libtool: compile:  clang++ -std=gnu++11 -DHAVE_CONFIG_H -I. -I../autoconf -I../include -I../src -I../include -I../src -g -O2 -c base/output/cl_output_hex.cc -o base/output/cl_output_hex.o >/dev/null 2>&1
make: *** [all-recursive] Error 1
